crates/ruff_linter/src/rules/ruff/rules/redundant_bool_literal.rs

```diff
@@ -0,0 +1,129 @@
+use ruff_diagnostics::{Diagnostic, Edit, Fix, FixAvailability, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::Expr;
+use ruff_python_semantic::analyze::typing::traverse_literal;
+use ruff_text_size::Ranged;
+
+use bitflags::bitflags;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for `Literal[True, False]` type annotations.
+///
+/// ## Why is this bad?
+/// The `bool` type has exactly two constant instances: `True` and `False`. Writing
+/// `Literal[True, False]` in the context of type annotations could be replaced by a
+/// bare `bool` annotations. Static type checkers such as [mypy] treat them as
+/// equivalent.
+///
+/// ## Example
+/// ```python
+/// Literal[True, False]
+/// Literal[True, False, "hello", "world"]
+/// ```
+///
+/// Use instead:
+/// ```python
+/// bool
+/// Literal["hello", "world"] | bool
+/// ```
+///
+/// ## Fix safety
+/// The fix for this rule is marked as unsafe:
+///
+/// - The type annotation might be intentional when the type checker used does not
+/// treat `bool` as equivalent when overloading boolean arguments with `Literal[True]`
+/// and `Literal[False]`, e.g. see [#14764] and [#5421].
+/// - `bool` is not strictly equivalent to `Literal[True, False]`, as `bool` is
+/// a subclass of `int`, and this rule might not apply if the type annotations are used
+/// in a numerical context.
```

Comment:
(arguably it's a type-checker bug if they don't understand that `<object of type Literal[True]> + <object of type Literal[True]> == <object of type Literal[2]>` in a numeric context. But hey.)